### PR TITLE
choice documentation and samples

### LINKED ISSUE #1623: Document Choice CRD
## What is changing? (Please include as many details as possible.)

Introducing a new CRD called Choice

## How will this impact our users?

They will be able to use a new CRD to easily specify multiple branches guarded by an optional predicate.

## In what release will this take happen (to the best of your knowledge)?
0.8?

## Context
Link to associated PRs or issues from other repos here.

1. https://github.com/knative/eventing/pull/1529

### PR DESCRIPTION
**Pull request by lionelvillard**
Fixes #1623 
## Proposed Changes

- Add documentation for the new Choice CRD
- Add multiple case sample
- Add mutual exclusivity sample

### CODE REVIEW

**Comment by lionelvillard** [comment]
@vaikas-google @n3wscott

**Comment by lionelvillard** [comment]
Can we try to get this in for 0.8?

File: docs/eventing/samples/choice/_index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Choice Example"
+linkTitle: "Choice"
+weight: 10
+type: "docs"
+---
```
Comment (samodell) [inline docs/eventing/samples/choice/_index.md]:
```suggestion
---

{{% readfile file="README.md" relative="true" markdown="true" %}}
```

Comment (samodell) [reply docs/eventing/samples/choice/_index.md]: This pulls the content from the README into the index file, so that it display properly on the site.

**Comment by samodell** [comment]
Taking a look now, @lionelvillard ! Thanks for writing this up.

File: docs/eventing/choice.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+1. Conditions which detail the overall Status of the Choice object
+1. IngressChannelStatus and CaseStatuses which convey the Status
+   of underlying `Channel` and `Subscription` resource  that are created as part of this Choice.
+1. AddressStatus which is exposed so that Choice can be used where Addressable
+   can be used. Sending to this address will target the `Channel` which is
+   fronting this Choice (same as `IngressChannelStatus`).
+
+## Examples
+
+The best is to learn how to use Choice through [examples](./samples/choice/README.md)
```
Comment (samodell) [inline docs/eventing/choice.md]:
```suggestion
Learn how to use Choice by following the [examples](./samples/choice/README.md)
```

File: docs/eventing/choice.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+### Choice Spec
+
+Choice has three parts for the Spec:
+
+1. `cases` defines the list of `filter` and `subscriber` pairs, one per
+    branch, and optionally a `reply` object. For each branch:
+    1. the `filter` is evaluated and when it returns an event the `subscriber` is executed.
+    Both `filter` and `subscriber` must be `Callable`.
+    2. the event returned by the `subscriber` is sent to the branch `reply` object. When
```
Comment (n3wscott) [inline docs/eventing/choice.md]: nit: `1.`

File: docs/eventing/samples/choice/_index.md (hunk 1 above)
Comment (samodell) [reply docs/eventing/samples/choice/_index.md]: Oh, I see you have these everywhere else they're needed; looks like this one just got skipped.

**Pull request merged**